src/neo/ProtocolSettings.cs

```diff
@@ -18,6 +18,7 @@ public class ProtocolSettings
         public int MemoryPoolMaxTransactions { get; }
         public uint MaxTraceableBlocks { get; }
         public IReadOnlyDictionary<string, uint> NativeActivations { get; }
+        public IReadOnlyList<uint> EmergencyMode { get; }
```

Comment:
It also need `StartBlock`, which is an index of a block that start the emergency mode.

Reply:
I thought to put the block who was under emergency, in order to alter only these blocks.

Reply:
I agree with Erik, from `StartBlock`, and I even dare saying that we also need some `EndBlock`... because by changing consensus during emergency, we should guarantee a certain time to update this consensus, and come back to "normal" (at `EndBlock`. And since this emergency may happen many times (see my comment below), we will need a list of PastEmergencyEvents, including which height each emergency started, ended and which forced consensus it lead to. This means a big responsibility for nodes (and node maintainers) willing to perform several emergencies, which definitely shouldn't happen in a stable network.

Reply:
> I thought to put the block who was under emergency, in order to alter only these blocks.

Now I got your idea @shargon... You plan on putting a list of all blocks in emergency... I think it works, but it's so much better to make a list of pairs <begin,end>... because each emergency may last many blocks (due to its uncertain nature), and  I guess we need some explicit `EmergencyOut`, for nodes to know when they should exit emergency and go back to "normal". I think we need this `out` on blocks, but maybe it's better to have this on ProtocolSettings as well, to have some very certain and predictable behavior of what happened and when it happened.

Reply:
I understand the pairs, but it could be slower to check all the pairs because we will need to iterate all the possible emergency pairs, if we have a single list, it will be faster because it could be a `hashSet`.

I think that we should be able to solve the emergency in one block, otherwise, we always can add a new entry in this list. What do you think?